src/main/scala/lambdaone/toolbox/sql/CrudPickSqlCollector.scala

```diff
@@ -0,0 +1,145 @@
+package lambdaone.toolbox.sql
+
+import cats._
+import cats.implicits._
+import doobie._
+import doobie.implicits._
+import lambdaone.kratia.collector.CollectorCRUD.{AllVotes, BoxData}
+import lambdaone.kratia.collector.{
+  BinaryProposal,
+  InfluenceAllocation,
+  Timestamp
+}
+import lambdaone.toolbox.CRUDPick
+
+object CrudPickSqlCollector {
+
+  type BinaryBoxData[A, D] = BoxData[A, BinaryProposal, D]
+
+  def init[F[_]: Monad](implicit xa: Transactor[F]) =
+    (sql"create table ballotbox_binary (id varchar(255), data varchar(255), closes_on number)".update.run
+      *> sql"create table votes_binary (box varchar(255), member varchar(255), proof varchar(255), yes number, no number )".update.run)
+      .transact(xa)
+
+  def tearDown[F[_]: Monad](implicit xa: Transactor[F]) =
+    (sql"drop table if exists ballotbox_binary".update.run
+      *> sql"drop table if exists votes_binary".update.run).transact(xa)
+
+  def apply[F[_]: Monad, A: Read: Put, D: Read: Put](
+    implicit xa: Transactor[F]
+  ) = new CRUDPick[F, A, BoxData[A, BinaryProposal, D]] {
+
+    /** Stores `a` with the chosen new unique reference */
+    override def create(a: BoxData[A, BinaryProposal, D], id: A): F[A] = {
+      sql"insert into ballotbox_binary (id, data, closes_on) values ($id, ${a.data}, ${a.closedOn})".update.run
+        .flatMap {
+          case 1 =>
```

Comment:
is this a partial function? what happens if it returns another number?

Reply:
I'd like to `raise` an error, but we only have a Monad. Porposal: make it MonadError.

Reply:
Yes we can get a `F: MonadError[Throwable, F]`, cat's `IO` has an instance of that one